certbot/tests/cli_test.py

```diff
@@ -30,15 +30,21 @@ def test_read_file(self):
             # However a relative path between two different drives is invalid. So we move to
             # self.tempdir to ensure that we stay on the same drive.
             os.chdir(self.tempdir)
-            rel_test_path = os.path.relpath(os.path.join(self.tempdir, 'foo'))
+            # The read-only filesystem introduced with macOS Catalina can break
+            # code using relative paths below. See
+            # https://bugs.python.org/issue38295 for another example of this.
+            # Eliminating any possible symlinks in self.tempdir before passing
```

Comment:
I would like to note here that `filesystem.realpath` does not have the exact same behavior than `os.path.realpath`: on Windows `os.path.realpath` is not called, and currently it then not capable of resolving symlinks in parent directories for this platform.

Obviously it does not apply to MacOS X, but could you be a little more precise in your comment, saying that `filesystem.realpath` invokes `os.path.realpath` on Unix-like platform, and so will clean up any possible symlink in the path?

I would like to avoid to imply that `filesystem.realpath` equals `os.path.realpath` until https://github.com/certbot/certbot/issues/7262 is resolved.